core/src/main/java/org/jdbi/v3/core/extension/ExtensionMetadata.java

```diff
@@ -109,6 +112,14 @@ public Set<Method> getExtensionMethods() {
         return methodHandlers.keySet();
     }
 
+    /**
+     * Returns a reference to a method that overrides {@link Object#finalize()} if it exists.
+     * @return An {@link Optional} containing a {@link Method} if a finalizer exists.
+     */
+    public Optional<Method> getFinalizer() {
```

Comment:
I originally had this as package-private so it is not part of the public API.
Do we really think this should be a fully supported part of the public API?